### PR TITLE
Use deterministic cloning

### PR DESCRIPTION
**Pull request by bbenligiray**
Closes https://github.com/api3dao/stakeable-vesting/issues/1

### CODE REVIEW

File: test/StakeableVestingFactory.sol.js [hunk 1]
```diff
@@ -101,61 +101,115 @@ describe('StakeableVestingFactory', function () {
...
+                        [
+                          roles.beneficiary.address,
+                          vestingParameters.startTimestamp,
+                          vestingParameters.endTimestamp,
+                          vestingParameters.amount,
+                        ]
+                      ),
+                      ethers.utils.keccak256(
+                        ethers.utils.hexConcat([
+                          '0x3d602d80600a3d3981f3363d3d373d3d3d363d73',
```
Comment (Ashar2shahid) [inline test/StakeableVestingFactory.sol.js:124]: some comments on the right would help understand what this `from` address is

**Review by Ashar2shahid — APPROVED** [review]

File: test/StakeableVestingFactory.sol.js [hunk 2]
```diff
@@ -105,6 +105,29 @@ describe('StakeableVestingFactory', function () {
...
```
Comment (Ashar2shahid) [inline test/StakeableVestingFactory.sol.js:124]: took a while to understand where these values came from

Comment (bbenligiray) [reply test/StakeableVestingFactory.sol.js]: Improved that a bit

**Comment by acenolaza** [comment]
Will we ever want to know the proxy contract address before deployment? If so then we should consider adding a pure function `getProxyAddress(salt)` that calls `Clones.predictDeterministicAddress()` 🤔

**Comment by bbenligiray** [comment]
@acenolaza Not needed, I demonstrated how that's done here https://github.com/api3dao/stakeable-vesting/blob/ee5302357a2d30bf3322f355a78b4ceffc6ba8b2/test/StakeableVestingFactory.sol.js#L110

**Pull request merged**